### PR TITLE
all examples + nyuncli run fix + package bumps

### PR DESCRIPTION
**Pull request by shwu-nyunai**
- fixes a bug in the command added for nyunzero-cli
- add tensorrtllm example
- updates text-gen package with nyuntam_adapt removed as a submodule in text-gen and updated tensorrtllm;

[_👀 preview_](https://github.com/nyunAI/nyuntam/tree/tensorrtllm_example/examples)

### CODE REVIEW

**Review by Arnav0400 — APPROVED** [review]

**Review by Akash-guna — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,6 +1,126 @@
...
+# Nyuntam
 
-For installations and usage checkout -
-- [nyuntam-text-generation](https://github.com/nyunAI/nyuntam-text-generation/blob/main/README.md)
-- [nyuntam-vision](https://github.com/nyunAI/nyuntam-vision/blob/main/README.md)
+**Nyuntam** is NyunAI's advanced suite designed to optimize, adapt, and accelerate a wide array of deep learning models across various domains. The repository is structured into several submodules, each targeting specific tasks:
+
+- **Nyuntam Text Generation**: Focuses on compressing large language models for text generation tasks.
+- **Nyuntam Vision**: Tailored for compressing and optimizing vision models.
+- **Nyuntam Adapt**: A robust module for fine-tuning and transfer learning across both vision and language models.
```
Comment (Arnav0400) [inline README.md:7]: lets also mention a few exampler algos for adapt something like - A robust module for fine-tuning and transfer learning across both vision and language models leveraging SoTA PEFT, full-finetuning and GPU parallelism.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,126 @@
...
+
+## Key Features
+
+- **State-of-the-Art Compression**: Includes techniques such as pruning, quantization, distillation, and more, to ensure model efficiency without compromising performance.
+- **Adaptation**: Fine-tune and adapt models to specific tasks and datasets using methods like LoRA, SSF, and others.
```
Comment (Arnav0400) [inline README.md:12]: Fine-tune and adapt vision and language models to specific tasks and datasets using methods like (Q)LoRA, (Q)SSF, and others.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,126 @@
...
+- **Multi-Platform Support**: Run experiments seamlessly on various platforms using Docker or virtual environments.
+- **CLI Integration**: Simplify your workflow with the NyunZero CLI.
```
Comment (Arnav0400) [inline README.md:14]: Lets add link to the CLI repo here

File: README.md [hunk 4]
```diff
@@ -1,6 +1,126 @@
...
+
+    ```bash
+    python3 -m venv {ENVIRONMENT_NAME}
+    source {ENVIRONMENT_NAME}/bin/activate
+    ```
+
+3. **Install the Required Packages**:
+
+    ```bash
+    pip install -r <submodule>/requirements.txt
```
Comment (Arnav0400) [inline README.md:66]: can we have this in detail here?

File: README.md [hunk 5]
```diff
@@ -1,6 +1,126 @@
...
+
+### Setting Up YAML Configuration Files
+
+Each submodule in Nyuntam requires a YAML file that defines all the necessary hyperparameters and configurations for running experiments. Examples and templates for these YAML files can be found within each submodule.
+
+- **Nyuntam Text Generation**:
+  - Configuration details for datasets and models can be found [here](https://nyunai.github.io/nyun-docs/dataset/).
+  - Scripts and example YAML files are available [here](https://github.com/nyunAI/nyuntam-text-generation/tree/main/scripts).
+
+- **Nyuntam Vision**:
```
Comment (Arnav0400) [inline README.md:104]: Can we add link similar to text generation for both vision and adapt.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,126 @@
...
+  - Example YAML configurations are provided in the repository. Details on setting up paths, compression algorithms, and other parameters can be found in the documentation.
+
+- **Nyuntam Adapt**:
```
Comment (Arnav0400) [inline README.md:107]: Need to update the link similar to text generation

File: examples/adapt/image_segmentation/README.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+## Configuration
+
+The following YAML file is used for setting up the experiment : 
+
+```yaml
+
+#DATASET
+JOB_SERVICE: Adapt
+CUSTOM_DATASET_PATH : "/custom_data/iccv09Data"
```
Comment (Arnav0400) [inline examples/adapt/image_segmentation/README.md:92]: is this path correct? custom_data as an absolute path seems incorrect.

File: examples/adapt/image_segmentation/README.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+LAST_LAYER_TUNING : True
+FULL_FINE_TUNING : False
+
+PEFT_METHOD : 'SSF'
+peft_type : 'SSF'
+
+
+Library :  'MMSEG'
+cuda_id : '0'
+OUTPUT_DIR : "/user_data/jobs/Adapt/IMGSEG"
```
Comment (Arnav0400) [inline examples/adapt/image_segmentation/README.md:155]: absolute paths for user_data also seem incorrect

File: examples/text-generation/aqlm_quantization/readme.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+  --tasks "gptq_wikitext:0" \
+  --results "user_data/evals/meta-llama_3.1-8b/Llama3Q_PV_Tuned" \
+  --cache_dir "user_data/.cache"
+```
+
+Compare the results with the original model to assess the impact of quantization on accuracy and inference speed.
+
+|                 | Llama3-8b  | Llama3Q PV Tuned  |
+|---------------- |----------- |------------------ |
+| GSM8K (5 shot)  | 50.9       | 58.9              |
```
Comment (Arnav0400) [inline examples/text-generation/aqlm_quantization/readme.md:243]: lets mention that we have used a fine-tuned llama3 and also include its number

File: examples/text-generation/flap_pruning/readme.md [hunk 10]
```diff
@@ -162,32 +165,27 @@ pip install lm-eval git+https://github.com/PanQiWei/AutoGPTQ.git
...
-| Model       	| Task                        	| Metric       	| Baseline 	| Pruned   	| Impact 	|
-|-------------	|-----------------------------	|--------------	|----------	|----------	|--------	|
-| Llama3.1-70b 	| Size (in GB)                	| Model Size ↓ 	| 132.0    	| 85.0     	| -47.0 	|
-| Llama3.1-70b 	| Perplexity (wikitext, gptq) 	| Perplexity ↓ 	| 3.79     	| 84146.23 	| > +84k 	|
-| Llama3.1-70b 	| ARC Challenge (25 shot)     	| Accuracy ↑   	| 70.48    	| 27.82    	| -42.66 	|
-
-<br>
+| Model         | Task                         | Metric        | Baseline  | Pruned    |
+| ------------- |----------------------------- |-------------- |---------- |---------- |
+| Llama3.1-70b  | MMLU (5 shot)                | Accuracy ↑    | 83.6      | 82.31     |
```
Comment (Arnav0400) [inline examples/text-generation/flap_pruning/readme.md:178]: lets also add num parameters reduction

**Review by Arnav0400 — CHANGES_REQUESTED** [review]

File: examples/text-generation/flap_pruning/readme.md [hunk 11]
```diff
@@ -1,17 +1,17 @@
-# FLAP 0.5x Llama3.1-70b
+# Llama3.1 70B: 0.5x the cost & size
```
Comment (Arnav0400) [inline examples/text-generation/flap_pruning/readme.md:1]: this will not be 0.5x now as start layer is 56

**Review by Arnav0400 — CHANGES_REQUESTED** [review]

File: examples/text-generation/flap_pruning/readme.md (hunk 11 above)
Comment (shwu-nyunai) [reply examples/text-generation/flap_pruning/readme.md:1]: will infer this based on the difference in parameters.

File: examples/adapt/image_segmentation/README.md (hunk 7 above)
Comment (shwu-nyunai) [reply examples/adapt/image_segmentation/README.md:92]: all paths in the cli flow are absolute paths. The "custom_data" path while creating the Nyun-workspace is mounted to "/custom_data" in the docker hence the path.

Comment (shwu-nyunai) [reply examples/adapt/image_segmentation/README.md:92]: Adapt -https://github.com/nyunAI/nyuntam_adapt/blob/228e5ec7394dc03d2d88fdda4ed3c85295310fd3/core/base_task.py#L72

Text-Gen - 
https://github.com/nyunAI/nyuntam-text-generation/blob/9542023ef0836e4346c45eecaad83711c71efadc/core/dataset.py#L60

Vision -
https://github.com/nyunAI/nyuntam-vision/blob/edcc1ff5e9061351efa2b974b0591b4d05df973b/core/models/__init__.py#L34

The custom paths are directly checked for existence. Without prefixing any parent custom_path (i.e. not considering as relative path) and hence the above.

File: examples/text-generation/aqlm_quantization/readme.md (hunk 9 above)
Comment (shwu-nyunai) [reply examples/text-generation/aqlm_quantization/readme.md:243]: as above

**Review by Arnav0400 — APPROVED** [review]
